gossip-protocol-spec.md

```diff
@@ -156,16 +156,23 @@ enum Protocol {
     //...
 }
 ```
-
 </details>
 
+The node shouldn't process crds values from the push message sent by node that hasn't yet replied to a recent [ping message](#ping-message).
+
 ### Pull request
 A node sends a pull request to ask the cluster for new information. It creates a set of bloom filters populated with the hashes of the `CrdsValue`s in its `crds` table and sends different bloom filters to different peers. The recipients of the pull request use the received bloom filter to identify what information the sender is missing and then construct a [pull response](#pull-response) packed with the missing `CrdsValue` data for the origin of the pull request.
 
 | Data | Type | Size | Description |
 |------|:----:|:----:|-------------|
 | `CrdsFilter` | [`CrdsFilter`](#crdsfilter) | 37+ | a bloom filter representing `CrdsValue`s the node already has |
-| `CrdsValue` | [`CrdsValue`](#data-shared-between-nodes) | ? | a value, usually a [`LegacyContactInfo`](#legacycontactinfo) of the node that sent the pull request |
+| `ContactInfo` | [`CrdsValue`](#data-shared-between-nodes) | ? | The `crds` value containing contact info of the node that sent the pull request |
+
+The only allowed values are [`ContactInfo`](#contactinfo) and [`LegacyContactInfo`](#legacycontactinfo) of the node that sent the pull request. The recipient node should not process this value and store it in the `CRDS` table - this value should be used only for the following:
+ - Use it to check that the node is not sending a pull request to itself
+ - Check whether the sender node responded to a `Ping` message. If the node still hasn't replied to Ping message, generate a `Ping` message for the sender node, unless the recipient node is already awaiting for the `Ping` response.
+
```

Comment:
I'm not sure about those 2 points - they are rather implementation details and are not a "spec".  I would move them to the implementation details doc.

Reply:
I rephrased it a bit, so it sounds like a recommendation.